### PR TITLE
Fixes #170536: Accept int padding value for F.pad

### LINKED ISSUE #170536: accept int padding value for F.pad
### 🚀 The feature, motivation and pitch

currently the `pad` function only accept float type `value`
```
pad(input: torch.Tensor, pad: list[int], mode: str = 'constant', value: Optional[float] = None) -> torch.Tensor
```

if I'm trying to pad a int64 typed tensor, it may give wrong result
```
>>> a = torch.arange(10)
>>> F.pad(a, (0, 1), mode='constant', value=torch.iinfo(torch.long).max)
tensor([                   0,                    1,                    2,
                           3,                    4,                    5,
                           6,                    7,                    8,
                           9, -9223372036854775808])
```

here is the expected result:
```
>>> a = torch.arange(10)
>>> b = F.pad(a, (0, 1), mode='constant', value=-1)
>>> torch.masked_fill(b, b == -1, torch.iinfo(torch.long).max)
tensor([                  0,                   1,                   2,
                          3,                   4,                   5,
                          6,                   7,                   8,
                          9, 9223372036854775807])
```

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jayzuccarelli**
## Summary
`F.pad(..., mode="constant", value=...)` previously treated `value` as a float. Padding an integer tensor (e.g. `int64`) with an int fill like `value=torch.iinfo(torch.long).max` was converted to double and then written back to the tensor, producing incorrect values (e.g. overflow).

This change threads a **Scalar** for the padding value from the Python API through to `constant_pad_nd`, so integer tensors can be padded with integer fill values correctly.

## Changes
- **native_functions.yaml**: `pad` and `_pad_enum` now use `Scalar? value=None` instead of `float? value=None`.
- **PadNd.cpp**: `_pad_enum_symint` and `pad_symint` take `std::optional<c10::Scalar>` and pass it through to `constant_pad_nd_symint`.
- **C++ API**: `torch::nn::functional::pad` and `PadFuncOptions` use `c10::Scalar` for the fill value.
- **Python**: `torch.nn.functional.pad` type hint updated to `value: Optional[Union[float, int]]`.
- **Test**: `test_constant_pad_nd_int_value` added to pad an int64 tensor with `torch.iinfo(torch.long).max` and assert the padded region is correct.

Fixes #170536

### CODE REVIEW

File: aten/src/ATen/native/PadNd.cpp [hunk 1]
```diff
@@ -204,17 +204,17 @@ static std::string_view padding_mode_string(padding_mode m) {
...
   TORCH_CHECK(static_cast<int64_t>(pad.size()) <= input_dim * 2,
               "Padding length should be less than or equal to two times the input dimension but got padding length ", pad.size(), " and input of dimension ", input_dim);
   auto mode = static_cast<at::padding_mode>(mode_int);
 
   if (mode == at::padding_mode::constant) {
-    return at::constant_pad_nd_symint(self, pad, value.value_or(0.0));
+    return at::constant_pad_nd_symint(self, pad, value.value_or(c10::Scalar(0)));
   }
-  TORCH_CHECK(!value.has_value() || *value == 0,
+  TORCH_CHECK(!value.has_value() || value->toDouble() == 0,
```
[the remark this answers is not in the document]
Comment (jayzuccarelli) [reply aten/src/ATen/native/PadNd.cpp]: FIxed
